### PR TITLE
Make `SecretHash` `Copy`

### LINKED ISSUE #440: Make rfc003::SecretHash implement Copy
Currently, the `SecretHash` implementation for `rfc003` is not Copy because it is based on a `Vec`. However, we know that it is always 32 bytes since it is based on `SHA256`.

We should refactor it to a be represented as `[u8; 32]` and make it implement `Copy`.

Note:
It is an array but just need to derive `Copy`.

DoD:
- [ ] derive `Copy` for `SecretHash`

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #440

---
- [x] RTM

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -204,6 +205,21 @@ private = true
 command = "sccache"
 args = ["--stop-server"]
 
+[tasks.check-evm]
```
Comment (LLFourn) [inline Makefile.toml]: So what's all this then?

**Comment by LLFourn** [comment]
@D4nte I think some unrelated changes ended up in this PR XD

File: Makefile.toml [hunk 2]
```diff
@@ -204,6 +205,21 @@ private = true
...
+# Ensure that when Travis compiled the EVM assembly
+# the hex result is in line with what was checked out
+description = "Ensure the EVM contract have not changed"
+workspace = false
+private = true
+script = [
+'''
+changes=$(git status -s |wc -l)
```
Comment (bonomat) [inline Makefile.toml]: does this work on Travis?

File: Makefile.toml (hunk 1 above)
Comment (D4nte) [reply Makefile.toml]: oopsie daysie

File: Makefile.toml (hunk 2 above)
Comment (D4nte) [reply Makefile.toml]: Yes, but it's in the wrong PR :)

**Pull request merged**